x/auth/vesting/types/codec.go

```diff
@@ -14,3 +16,27 @@ func RegisterCodec(cdc *codec.Codec) {
 	cdc.RegisterConcrete(&DelayedVestingAccount{}, "cosmos-sdk/DelayedVestingAccount", nil)
 	cdc.RegisterConcrete(&PeriodicVestingAccount{}, "cosmos-sdk/PeriodicVestingAccount", nil)
 }
+
+func RegisterInterfaces(registry types.InterfaceRegistry) {
+	registry.RegisterInterface(
```

Comment:
Why are we calling this twice?

Reply:
We need to registerInterfaces of both concrete types from auth and vesting. I tried doing it from auth codec but results in cyclic dependency and also violates the above said idea of auth not depending on vesting

Reply:
Ohhh I see, one calls it with `VestingAccount` and the other with `AccountI` -- very hard to spot :-p. Maybe add a quick comment?

Reply:
Sure